File: app/src/main/java/mohtasham/paydar/sabalan/ezbazi/controller/adapter/recyclerview/user_activity/ListShopRequestAdapter.java
package mohtasham.paydar.sabalan.ezbazi.controller.adapter.recyclerview.user_activity;

import android.annotation.SuppressLint;
import android.content.Context;
import android.content.Intent;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.TextView;

import com.makeramen.roundedimageview.RoundedImageView;
import com.squareup.picasso.Picasso;

import java.util.ArrayList;
import java.util.List;

import mohtasham.paydar.sabalan.ezbazi.R;
import mohtasham.paydar.sabalan.ezbazi.controller.system.HelperDate;
import mohtasham.paydar.sabalan.ezbazi.controller.system.HelperText;
import mohtasham.paydar.sabalan.ezbazi.model.ShopRequest;
import mohtasham.paydar.sabalan.ezbazi.model.common.Photo;
import mohtasham.paydar.sabalan.ezbazi.view.activity.ActivityShowShop;
import mohtasham.paydar.sabalan.ezbazi.view.custom_views.my_views.MyViews;

public class ListShopRequestAdapter extends RecyclerView.Adapter<ListShopRequestAdapter.ListViewHolder>{


  private Context context;
  private List<ShopRequest> requests;

  public ListShopRequestAdapter(Context context, List<ShopRequest> requests){

    this.context = context;
    this.requests = requests;
  }

  @Override
  public ListViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
    View view= LayoutInflater.from(context).inflate(R.layout.item_user_shop_request,parent,false);
    return new ListViewHolder(view);
  }

  @SuppressLint("SetTextI18n")
  @Override
  public void onBindViewHolder(ListViewHolder holder, int position) {
    final ShopRequest request = requests.get(position);
    ArrayList<Photo> photos = request.getGame().getPhotos();
    Picasso.with(context).
      load(photos.get(0).getUrl())
//      .noFade()
//      .memoryPolicy(MemoryPolicy.NO_CACHE, MemoryPolicy.NO_STORE)
//      .skipMemoryCache()
      //.placeholder(context.getResources().getDrawable(R.drawable.default_good_image)).
      //.error(context.getResources().getDrawable(R.drawable.default_no_image))
      .into(holder.img_game);

    holder.txt_name.setText(request.getGame().getName());
//      holder.txt_release_date.setText("تاریخ ارائه : " + game.getProduction_date());
//      holder.txt_rating.setText("rate : " + "4.5");
    holder.txt_region.setText("Region : " + request.getGame().getRegion());

    holder.btn_price.setText(HelperText.split_price(request.getGame_price()) + " تومان ");

    if(request.getIs_finish() == 1) {
      holder.txt_status.setText("تمام شده");
    }else if(request.getIs_finish() == 0 && request.getIs_success() == 1){
      holder.txt_status.setText("دریافت شده");
    }
    else if(request.getIs_finish() == 0 && request.getIs_success() == 0 && request.getIs_sent() == 1){
      holder.txt_status.setText("ارسال شده");
    }else if(request.getIs_finish() == 0 && request.getIs_success() == 0 && request.getIs_sent() == 0){
      holder.txt_status.setText("در حال ارسال");
    }

    holder.txt_date.setText(new HelperDate().timestampToPersian(request.getCreated_at()));






    holder.itemView.setOnClickListener(new View.OnClickListener() {
      @Override
      public void onClick(View v) {
        Intent intent = new Intent(context, ActivityShowShop.class);
        intent.putExtra("ID", request.getGame().getId());
        context.startActivity(intent);
      }
    });


  }

  @Override
  public int getItemCount() {
    return requests.size();
  }

  public class ListViewHolder extends RecyclerView.ViewHolder{

    private RoundedImageView img_game;
    private TextView txt_name;
    private TextView txt_rating;
    private TextView txt_region;
    private TextView txt_release_date;
    private Button btn_price;
    private TextView txt_show_status, txt_status, txt_date, txt_show_date;

    public ListViewHolder(View itemView) {
      super(itemView);
      img_game = itemView.findViewById(R.id.img_game);
      txt_name = itemView.findViewById(R.id.txt_name);
      txt_rating = itemView.findViewById(R.id.txt_rating);
      txt_region = itemView.findViewById(R.id.txt_region);
      txt_release_date = itemView.findViewById(R.id.txt_release_date);
      btn_price = itemView.findViewById(R.id.btn_price);
      txt_show_status = itemView.findViewById(R.id.txt_show_status);
      txt_status = itemView.findViewById(R.id.txt_status);
      txt_date = itemView.findViewById(R.id.txt_date);
      txt_show_date = itemView.findViewById(R.id.txt_show_date);


      setTypeFace();
    }

    private void setTypeFace(){
      txt_name.setTypeface(MyViews.getIranSansLightFont(context));
      txt_rating.setTypeface(MyViews.getIranSansLightFont(context));
      txt_region.setTypeface(MyViews.getIranSansLightFont(context));
      txt_release_date.setTypeface(MyViews.getIranSansMediumFont(context));
      btn_price.setTypeface(MyViews.getIranSansLightFont(context));

      txt_show_status.setTypeface(MyViews.getIranSansLightFont(context));
      txt_status.setTypeface(MyViews.getIranSansLightFont(context));
      txt_date.setTypeface(MyViews.getIranSansLightFont(context));
      txt_show_date.setTypeface(MyViews.getIranSansLightFont(context));
    }


  }


  public void notifyData(List<ShopRequest> requests1) {
    //Log.d("notifyData ", myList.size() + "");
    if (requests1.size() > 0) {
      for (int i = 0; i < requests1.size(); i++) {
        this.requests.add(requests1.get(i));
      }
      this.notifyItemInserted(requests.size() - 1);

      //notifyDataSetChanged();
    }
  }


  public void clear() {
    final int size = requests.size();
    requests.clear();
    notifyItemRangeRemoved(0, size);
  }







}

File: app/src/main/java/ir/easybazi/recyclerview_adapter/user_activity/ListShopRequestAdapter.kt
package ir.easybazi.recyclerview_adapter.user_activity

import android.annotation.SuppressLint
import android.content.Context
import android.content.Intent
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.makeramen.roundedimageview.RoundedImageView
import com.squareup.picasso.Picasso
import ir.easybazi.R
import ir.easybazi.model.ShopRequest
import ir.easybazi.system.helper.HelperDate
import ir.easybazi.system.helper.HelperText
import ir.easybazi.view.activity.ActivityShowShop
import ir.easybazi.view.custom_views.my_views.MyViews

class ListShopRequestAdapter(private val context: Context, private val requests: MutableList<ShopRequest>) : RecyclerView.Adapter<ListShopRequestAdapter.ListViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ListViewHolder {
        val view = LayoutInflater.from(context).inflate(R.layout.item_user_shop_request, parent, false)
        return ListViewHolder(view)
    }

    @SuppressLint("SetTextI18n")
    override fun onBindViewHolder(holder: ListViewHolder, position: Int) {
        val request = requests[position]
        if(request.game == null) return;
        if(request.game!!.app_cover_photo_url!!.length > 0) {
            val image_url = request.game!!.app_cover_photo_url
            Picasso.with(context).load(image_url)
                    //      .noFade()
                    //      .memoryPolicy(MemoryPolicy.NO_CACHE, MemoryPolicy.NO_STORE)
                    //      .skipMemoryCache()
                    //.placeholder(context.getResources().getDrawable(R.drawable.default_good_image)).
                    //.error(context.getResources().getDrawable(R.drawable.default_no_image))
                    .into(holder.img_game)
        }

        holder.txt_name.text = request.game!!.name
        //      holder.txt_release_date.setText("تاریخ ارائه : " + game.getProduction_date());
        //      holder.txt_rating.setText("rate : " + "4.5");
        holder.txt_region.text = "Region : " + request.game!!.region!!

        holder.btn_price.text = HelperText.split_price(request.game_price) + " تومان "

        if (request.is_finish === 1) {
            holder.txt_status.text = "تمام شده"
        } else if (request.is_finish === 0 && request.is_delivered === 1) {
            holder.txt_status.text = "دریافت شده"
        } else if (request.is_finish === 0 && request.is_delivered === 0 && request.is_sent === 1) {
            holder.txt_status.text = "ارسال شده"
        } else if (request.is_finish === 0 && request.is_delivered === 0 && request.is_sent === 0) {
            holder.txt_status.text = "در حال ارسال"
        }

        holder.txt_date.text = HelperDate().timestampToPersian(request.created_at!!)






        holder.itemView.setOnClickListener {
            val intent = Intent(context, ActivityShowShop::class.java)
            intent.putExtra("ID", request.game!!.id)
            context.startActivity(intent)
        }


    }

    override fun getItemCount(): Int {
        return requests.size
    }

    inner class ListViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {

        internal val img_game: RoundedImageView
        internal val txt_name: TextView
        internal val txt_rating: TextView
        internal val txt_region: TextView
        internal val txt_release_date: TextView
        internal val btn_price: Button
        internal val txt_show_status: TextView
        internal val txt_status: TextView
        internal val txt_date: TextView
        internal val txt_show_date: TextView

        init {
            img_game = itemView.findViewById(R.id.img_game)
            txt_name = itemView.findViewById(R.id.txt_name)
            txt_rating = itemView.findViewById(R.id.txt_rating)
            txt_region = itemView.findViewById(R.id.txt_region)
            txt_release_date = itemView.findViewById(R.id.txt_release_date)
            btn_price = itemView.findViewById(R.id.btn_price)
            txt_show_status = itemView.findViewById(R.id.txt_show_status)
            txt_status = itemView.findViewById(R.id.txt_status)
            txt_date = itemView.findViewById(R.id.txt_date)
            txt_show_date = itemView.findViewById(R.id.txt_show_date)


            setTypeFace()
        }

        private fun setTypeFace() {
            txt_name.typeface = MyViews.getRobotoLightFont(context)
            txt_rating.typeface = MyViews.getRobotoLightFont(context)
            txt_region.typeface = MyViews.getRobotoLightFont(context)
            txt_release_date.typeface = MyViews.getIranSansMediumFont(context)
            btn_price.typeface = MyViews.getIranSansLightFont(context)

            txt_show_status.typeface = MyViews.getIranSansLightFont(context)
            txt_status.typeface = MyViews.getIranSansLightFont(context)
            txt_date.typeface = MyViews.getIranSansLightFont(context)
            txt_show_date.typeface = MyViews.getIranSansLightFont(context)
        }


    }


    fun notifyData(requests1: List<ShopRequest>) {
        //Log.d("notifyData ", myList.size() + "");
        if (requests1.size > 0) {
            for (i in requests1.indices) {
                this.requests.add(requests1[i])
            }
            this.notifyItemInserted(requests.size - 1)

            //notifyDataSetChanged();
        }
    }


    fun clear() {
        val size = requests.size
        requests.clear()
        notifyItemRangeRemoved(0, size)
    }


}